spec/system/grant_submissions/grant_submission_forms_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'rails_helper'
+
+RSpec.describe 'GrantSubmissiion::Forms', type: :system do
+  add_question_text = 'Add a Question to this Section'
+  add_section_text  = 'Add a Section'
+
+  describe 'Draft Grant Edit form', js: true do
+    before(:each) do
+      @draft_grant = create(:draft_grant_with_users) # includes one question
+      @form        = @draft_grant.form
+      @original_first_question  = @draft_grant.questions.find_by(display_order: 1)
+      @original_second_question = @draft_grant.questions.find_by(display_order: 2)
+      @original_third_question  = @draft_grant.questions.find_by(display_order: 3)
+
+      login_as(@draft_grant.grant_permissions.role_admin.first.user)
+      visit edit_grant_form_path(@draft_grant, @form)
+    end
+
+    scenario 'has edit links' do
+      expect(page).to have_link add_section_text
+      expect(page).to have_button 'Save'
+    end
+
+    context 'section' do
+      scenario 'can be added to a form' do
+        expect do
+          new_section_text = Faker::Lorem.sentence
+          click_link add_section_text
+          find_field('Title', with: '').set(new_section_text)
+          click_button 'Save'
+        end.to (change{@draft_grant.form.sections.count}).by 1
+        expect(page).to have_text 'Submission Form successfully updated'
+      end
+
+      scenario 'display_order is updated' do
+        original_section = @draft_grant.form.sections.first
+        expect(original_section.display_order).to be 1
+        find('.delete-section').click
+        click_link add_section_text
+        new_section_text = Faker::Lorem.sentence
+        find_field('Title', with: '').set(new_section_text)
+        click_button 'Save'
+        expect(@draft_grant.form.sections.first.display_order).not_to be 2
```

Comment:
sections.first is not necessarily the one that should be assigned display_order 2.
the section to check should be the one that had display_order 2 before the edit.